### PR TITLE
Update annotation vector styler

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr introduces many changes among features and bugfixes.
- the annotation styler UI has changed.
- each Feature can have an array of styles inside the style property
- support for symbols (svg) has been introduced, and they can be customized
- lines styler has three styles section

## Issues
 - no issue associated

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [x] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
you cannot:
- specify a symbol for the point annotation
- customize start end point

**What is the new behavior?**
you can:
- specify a symbol style for point
- customize start end point

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 1]
```diff
@@ -155,9 +169,16 @@ class DrawSupport extends React.Component {
...
         }
+        let newFeature = head(newProps.features);
+        if (newFeature && newFeature.features && newFeature.features.length) {
+            // filtering circles features only when drawing
+            return this.addFeatures(assign({}, newProps, {features: [{...newFeature, features: newFeature.features.filter(f => !f.properties.isCircle || !newProps.options.drawEnabled)}]}));
+        }
+        return this.addFeatures(newProps);
 
+/*
         const feature = this.addFeatures(newProps);
```
Comment (offtherailz) [inline web/client/components/map/openlayers/DrawSupport.jsx]: remove commented code

File: web/client/components/map/openlayers/VectorStyleNew.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
+const {isNil, trim, isString, isArray, castArray, head, last, find} = require('lodash');
```
Comment (offtherailz) [inline web/client/components/map/openlayers/VectorStyleNew.js]: Rename this VectorStyle, if you have to keep the old one, rename it Legacy, or something like that

File: web/client/utils/openlayers/StyleUtils.js [hunk 3]
```diff
@@ -38,6 +39,7 @@ const toVectorStyle = function(layer, style) {
         switch (getGeomType(layer)) {
         case 'Polygon':
         case 'MultiPolygon': {
+            // TODO clear this, it goes in maximum call stack size exceeded
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/utils/openlayers/StyleUtils.js]: this is there from ages. i think we can open a separate issue to fix this
@tdipisa opinions?

File: web/client/components/style/vector/marker/SymbolLayout.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        onChange: PropTypes.func,
+        onUpdateOptions: PropTypes.func,
+        width: PropTypes.number,
+        symbolsPath: PropTypes.string
+    };
+
+    static defaultProps = {
+        symbolsPath: "product/assets/symbols/",
+        style: {},
+        shape: "stop-hexagonal-signal",
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/components/style/vector/marker/SymbolLayout.jsx]: a flag? bring me an svg

**Pull request merged**